src/Draggable/Emitter/Emitter.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * The Emitter is a simple emitter class that provides you with `on()`, `off()` and `trigger()` methods
+ * @class Emitter
+ * @module Emitter
+ */
+export default class Emitter {
+  constructor() {
+    this.callbacks = {};
+  }
+
+  /**
+   * Registers callbacks by event name
+   * @param {String} type
+   * @param {...Function} callbacks
+   */
+  on(type, ...callbacks) {
+    if (!this.callbacks[type]) {
+      this.callbacks[type] = [];
+    }
+
+    this.callbacks[type].push(...callbacks);
+
+    return this;
+  }
+
+  /**
+   * Unregisters callbacks by event name
+   * @param {String} type
+   * @param {Function} callback
+   */
+  off(type, callback) {
+    if (!this.callbacks[type]) {
+      return null;
+    }
+
+    const copy = this.callbacks[type].slice(0);
```

Comment:
You just want to copy the array? Why not do `const copy = [...this.callbacks[type]];` instead of `slice(0)`? Personally I find the spread easier to understand. I might not understand this correctly however... if this is a deeply nested array, perhaps the spread behaves differently than I expect... never actually tried with a multi-level array.

Regardless, `slice` kind of implies you are a pro javascripter who has encountered this trick before - where as spreading out the array seems to more explicitly say "copy this value to this new variable".

Reply:
That is a good point 👍Will address this in a quick follow-up PR, thanks for mentioning!